packages/forms-web-app/src/dynamic-forms/dynamic-components/multi-file-upload/question.js

```diff
@@ -84,15 +85,62 @@ class MultiFileUploadQuestion extends Question {
 	prepQuestionForRendering(section, journey, customViewData) {
 		const viewModel = super.prepQuestionForRendering(section, journey, customViewData);
 
-		const answer = journey.response.answers[this.fieldName] || '';
+		const uploadedFiles = journey.response.answers.SubmissionDocumentUpload || [];
 
-		if (answer.uploadedFiles) {
-			viewModel.uploadedFiles = answer.uploadedFiles;
+		const relevantUploadedFiles = uploadedFiles.filter(
+			(upload) => upload.type === this.documentType.name
+		);
+
+		if (relevantUploadedFiles.length > 0) {
+			viewModel.uploadedFiles = relevantUploadedFiles;
 		}
 
 		return viewModel;
 	}
 
+	/**
+	 * Save the answer to the question
+	 * @param {ExpressRequest} req
+	 * @param {ExpressResponse} res
+	 * @param {Journey} journey
+	 * @param {Section} section
+	 * @param {JourneyResponse} journeyResponse
+	 * @returns {Promise<void>}
+	 */
+	async saveAction(req, res, journey, section, journeyResponse) {
+		// check for validation errors
+		const errorViewModel = this.checkForValidationErrors(req, section, journey);
+		if (errorViewModel) {
+			return this.renderAction(res, errorViewModel);
+		}
+
+		// save
+		const { uploadedFiles } = await this.getDataToSave(req, journeyResponse);
+		await uploadedFiles.forEach(async (file) => {
+			const data = {
+				...file,
+				type: this.documentType.name
+			};
+			await apiClient.postSubmissionDocumentUpload(journeyResponse.referenceId, data);
+		});
```

Comment:
Let's go with something like:

`await Promise.all(uploadedFiles.map((file) => {
  const data = {
    ...file,
    type: this.documentType.name
  };
  return apiClient.postSubmissionDocumentUpload(journeyResponse.referenceId, data);
});
```

we don't need to wait for each in turn.

Possible future improvement: the API could accept an array.

Reply:
See https://github.com/Planning-Inspectorate/appeal-planning-decision/pull/3404/commits/88d69b26cace909c16a75a6b7cf09898cb96dd38